src/main/java/net/dancier/domain/Recommendation.java

```diff
@@ -0,0 +1,32 @@
+package net.dancier.domain;
+
+import net.dancier.domain.dance.Dancer;
+import net.dancier.domain.dance.School;
+
+public class Recommendation<T extends Recommendable> {
+
+    public enum Type {
+        DANCER,
+        SCHOOL,
+        WORKSHOP,
+        EVENING,
+        ETC
+    }
+
+    public Type type;
+    public T payload;
+
+    public static Recommendation of(Recommendable recommendable) {
```

Comment:
I suggest extending the enum with one or more class references for auto resolving instead of manual mappping.

Reply:
I a currently wondering, if we need this wrapper at all. So give me some time. When we will stick to this solution with recommendation, I will consider your suggestion.

Reply:
Ok, I think we still need this wrapper.
What do you mean with the adding a class reference to the enum:
Dancer("dancer", Dancer.class) ?
How do I get autoresolving then?